### PR TITLE
[tools] Fix `LLVM_USE_SANITIZE_COVERAGE` build

### PR DESCRIPTION
**Pull request by justinfargnoli**

### CODE REVIEW

**Comment by justinfargnoli** [comment]
Ping for review @nikic @labrinea

File: llvm/tools/llvm-dis-fuzzer/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 set(LLVM_LINK_COMPONENTS
   BitReader
+  Core
+  Support
```
Comment (nikic) [inline llvm/tools/llvm-dis-fuzzer/CMakeLists.txt:4]: Why is this necessary? They should already be transitive deps of BitReader: https://github.com/llvm/llvm-project/blob/be10070f91b86a6f126d2451852242bfcb2cd366/llvm/lib/Bitcode/Reader/CMakeLists.txt#L14

Comment (justinfargnoli) [reply llvm/tools/llvm-dis-fuzzer/CMakeLists.txt:4]: To be frank, I don't have a strong enough understanding of the LLVM build system to properly articulate an answer. 

This change was necessary to get my local build working. 

My local build uses `-DBUILD_SHARED_LIBS=ON`.  When I tested `-DBUILD_SHARED_LIBS=OFF`, this change was not needed to use `-DLLVM_USE_SANITIZE_COVERAGE=ON`.

So, perhaps the transitive dependencies are not automatically added when using `-DBUILD_SHARED_LIBS=ON`. Does this seem reasonable to you @nikic?

Comment (nikic) [reply llvm/tools/llvm-dis-fuzzer/CMakeLists.txt:4]: Hm yeah, after looking at the cmake code a bit I think we make the dependencies of shared objects private.

**Review by nikic — APPROVED** [review]
LGTM
